seamm_datastore/api.py

```diff
@@ -81,148 +290,145 @@ def add_user(
-        roles = ["user"]
-
-    if groups is None:
-        groups = ["staff"]
+    """
+    Add a new user to the database.
 
-    # Verify username and password
-    # Check if user exists
-    from seamm_datastore.database.models import User
+    Parameters
+    ----------
+    session : sqlalchemy.Session
+        The session used to access the database.
+    username : str
+        The username that identifies the user.
+    password : str
+        The secret password for the user.
+    first_name : str = None
+        The user's first (given) name.
+    last_name : str = None
+        The user's last (family) name.
+    email : str = None
+        The user's principal email address.
+    roles : [str] = ["user"]
+        A list of roles for the user. Defaults to just "user".
+    groups : [str] = ["staff"]
+        A list of groups that the user belongs to. Defaults to "staff".
+    as_json : bool = False
+        If True, return the json description of the user; otherwise, the user id.
+    current_user : str or User = None
+        The user currently logged in. Not used.
+    """
+    from seamm_datastore.database.models import Group, Role, User
     from seamm_datastore.database.schema import UserSchema
 
+    # Check if the user already exists.
     user = User.query.filter_by(username=username).one_or_none()
-
     if user:
         raise ValueError(f"User {user} already found in the database")
 
+    # Create the new user and store in the database
     new_user = User(
         username=username,
         password=password,
         first_name=first_name,
         last_name=last_name,
         email=email,
-        roles=roles,
-        groups=groups,
     )
-
+    for role_name in roles:
+        role = Role.query.filter_by(name=role_name).one()
+        new_user.roles.append(role)
+    for group_name in groups:
+        group = Group.query.filter_by(name=group_name).one()
+        new_user.groups.append(group)
     session.add(new_user)
     session.commit()
 
+    # Return the json or user, as requested.
     if as_json:
-        new_user = UserSchema().dump(new_user)
-
-    return new_user
-
-
-def add_flowchart(session, flowchart_info):
-    from seamm_datastore.database.models import Flowchart
-
-    try:
-        flowchart = Flowchart.query.filter_by(
-            sha256_strict=flowchart_info["sha256_strict"]
-        ).one_or_none()
-    except KeyError:
-        try:
-            flowchart = Flowchart.query.filter_by(id=flowchart_info["id"]).one_or_none()
-        except KeyError:
-            flowchart = None
-
-    if flowchart:
-        raise ValueError(f"Flowchart already in datastore. ID: {flowchart.id}")
-
-    new_flowchart = Flowchart(**flowchart_info)
-
-    session.add(new_flowchart)
-    session.commit()
-
-    return new_flowchart
+        return UserSchema().dump(new_user)
+    else:
+        return new_user
 
 
-def add_job(session, job_data, as_json=False):
-    """Add a job to the datastore.
+def finish_job(
```

Comment:
I think this should be a general `update_job` function where you pass in appropriate arguments so that the status is set to "finished".